ml-agents/mlagents/trainers/settings.py

```diff
@@ -588,9 +600,17 @@ def structure(d: Mapping, t: type) -> Any:
         Helper method to structure a TrainerSettings class. Meant to be registered with
         cattr.register_structure_hook() and called with cattr.structure().
         """
+
         if not isinstance(d, Mapping):
             raise TrainerConfigError(f"Unsupported config {d} for {t.__name__}.")
+
         d_copy: Dict[str, Any] = {}
+
+        # Check if a default_settings was specified. If so, used those as the default
+        # rather than an empty dict.
+        if TrainerSettings.default_override is not None:
+            d_copy.update(cattr.unstructure(TrainerSettings.default_override))
```

Comment:
I'm not sure if this will do what you want for nested dictionaries. For example if you had 
```
{"foo": {"x": 1}}.update({"foo", {"y": 2}})
```
you'd end up with `{"foo", {"y": 2}}`, but I think you really want `{"foo", {"x": 1, "y": 2}}` (i.e. recursively merge dictionary values)

Reply:
Fixed using a recursive dict update